GitCommands/Git/GitModule.cs

```diff
@@ -1004,17 +1004,21 @@ string ProcessDiffNotes(int startIndex)
             }
         }
 
-        public IReadOnlyList<ObjectId> GetParents(ObjectId commitId)
+        public IReadOnlyList<ObjectId> GetParents(ObjectId objectId)
         {
-            GitArgumentBuilder args = new("log")
+            if (objectId.IsArtificial)
             {
-                "-n 1",
-                "--format=format:%P",
-                commitId
+                // For WorkTree could Index be returned, but for Index HEAD should be available anyway
+                throw new ArgumentException(nameof(objectId));
```

Comment:
I think InvalidOperationException may be more appropriate

Reply:
Will change and rebase merge.